src/sentry/services/hybrid_cloud/region.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from sentry.models import OrganizationMapping
+from sentry.services.hybrid_cloud import ArgumentDict
+from sentry.services.hybrid_cloud.rpc import RpcServiceSetupException
+from sentry.types.region import Region, get_region_by_name
+
+
+class RegionResolution(ABC):
+    @abstractmethod
+    def resolve(self, arguments: ArgumentDict) -> Region:
+        raise NotImplementedError
+
+    def _resolve_from_mapping(self, mapping: OrganizationMapping) -> Region:
+        return get_region_by_name(mapping.region_name)
+
+
+@dataclass
+class ByOrganizationObject(RegionResolution):
+    parameter_name: str | None = None
+
+    def resolve(self, arguments: ArgumentDict) -> Region:
+        from sentry.services.hybrid_cloud.organization import RpcOrganizationSummary
+
+        if self.parameter_name is None:
```

Comment:
Should this be an error? Services that need to resolve with an organization instance could provide the parameter name so that there is no guessing.

Reply:
Could go either way. My motivation was to make the default case terser (i.e., `resolve=ByOrganizationObject()` over `resolve=ByOrganizationObject("organization")` where the former is unambiguous) but I'm fine with taking the inference out of it.